src/Collection/CustomTypeCollection.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace BestIt\CTCustomTypesBundle\Collection;
+
+/**
+ * Provide a collection for custom type keys
+ * @author chowanski <chowanski@bestit-online.de>
+ * @package BestIt\CTCustomTypesBundle
+ * @subpackage Collection
+ * @version $id$
+ */
+class CustomTypeCollection
+{
+    /**
+     * The collection
+     * @var array
+     */
+    private $collection;
```

Comment:
You start with void and not array. it still could be void, even on the first get or has call.